### ISSUE TITLE
Build with pnpmEnable is failing with 'getClosestFocusable' is not exported

### ISSUE THREAD

**Issue by davidef** [vaadin/flow#18149]
### Description of the bug

Since this morning CI build (23.3.28) with pnpmEnable without any code change from our side started failing with the following error:
Error: 'getClosestFocusable' is not exported by node_modules/@vaadin/component-base/src/focus-utils.js, imported by node_modules/@vaadin/overlay/src/vaadin-overlay.js

The function referenced seems that have been added in a newer version compared to the one we are building with:
https://github.com/vaadin/web-components/commit/ee01a1ae03e28b38778789d2ecbc95f426ddeab1
The issue seems to have started with the release of updated npm packages here: https://www.npmjs.com/package/@vaadin/component-base?activeTab=versions

The same code without pnpmEnable is building fine

### Expected behavior

Project build both with pnpmEnable=true and pnpmEnable=false

### Minimal reproducible example

We used a clean container maven:3.9.5-eclipse-temurin-17)
From vaadin start -> flow 23.3.28 + java17 + maven servlet
Edit pom.xml adding `<pnpmEnable>true</pnpmEnable>` in vaadin-maven-plugin configuration
mvn -P production clean package

### Versions

- Vaadin / Flow version: 23.3.28
- Java version: 17

**Comment by NikolajOlejnik** [comment]
We have the same problem, 23.3.25.

**Comment by web-padawan** [comment]
The root cause is the fact that `@vaadin/overlay` transitive dependency isn't pinned with PNPM, for example:

```
  /@vaadin/dialog@23.3.17:
    resolution: {integrity: sha512-Qc+F+Qc7q8bEDTKy+DB39EXiuuCUeXih6sS9lA6UTBMAaH182crJt/vQ4RKlCIIch4a+QFHmLhxoMHz8ZfX5bg==}
    dependencies:
      '@open-wc/dedupe-mixin': 1.4.0
      '@polymer/polymer': 3.5.1
      '@vaadin/component-base': 23.3.17
      '@vaadin/lit-renderer': 23.3.17
      '@vaadin/overlay': 23.3.28
      '@vaadin/vaadin-lumo-styles': 23.3.17
      '@vaadin/vaadin-material-styles': 23.3.17
      '@vaadin/vaadin-themable-mixin': 23.3.17
```

**Comment by TGit911iger** [comment]
Same Problem here. Had to change our pipelines to npm. Waiting for fixing.

**Comment by sm7124** [comment]
@web-padawan In which version this fix will be available. We are using 23.3.25 and facing the same issue.

**Comment by mshabarov** [comment]
@sm7124 the fix should be available in the next Vaadin 23.3.29 version.

**Comment by mshabarov** [comment]
Closed, as the patch has been merged.
